etc/generate-envoy-config/envoy.libsonnet

```diff
@@ -102,79 +198,80 @@
+    listener_filters: [
+      {
+        name: 'envoy.filters.listener.tls_inspector',
+        typed_config: {
+          '@type': 'type.googleapis.com/envoy.extensions.filters.listener.tls_inspector.v3.TlsInspector',
+        },
+      },
+    ],
+    filter_chains: [
+      {
+        filter_chain_match: {
+          transport_protocol: 'tls',
+        },
+        transport_socket: {
+          name: 'envoy.transport_sockets.tls',
+          typed_config: {
+            '@type': 'type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.DownstreamTlsContext',
+            common_tls_context: {
+              alpn_protocols: ['h2', 'http/1.1'],
+              tls_params: {
+                tls_minimum_protocol_version: 'TLSv1_2',
+                cipher_suites: [
+                  '[ECDHE-ECDSA-AES128-GCM-SHA256|ECDHE-ECDSA-CHACHA20-POLY1305]',
+                  '[ECDHE-RSA-AES128-GCM-SHA256|ECDHE-RSA-CHACHA20-POLY1305]',
+                  'ECDHE-ECDSA-AES256-GCM-SHA384',
+                  'ECDHE-RSA-AES256-GCM-SHA384',
+                ],
               },
-              http_filters: std.prune([
-                // If there's a GRPC router, enable grpc_stats.
-                if std.foldl(function(last, route) last || 'grpc' in route.match, routes, false) then
-                  {
-                    name: 'envoy.filters.http.grpc_stats',
-                    typed_config: {
-                      '@type': 'type.googleapis.com/envoy.extensions.filters.http.grpc_stats.v3.FilterConfig',
-                      enable_upstream_stats: true,
-                      // Stats for all methods = true would let anyone on the Internet fill up the
-                      // statistics buffer with bogus RPC method names, so we have to leave it off.
-                      stats_for_all_methods: false,
-                    },
-                  } else {},
+              tls_certificate_sds_secret_configs: [
                 {
-                  name: 'envoy.filters.http.router',
-                  typed_config: {
-                    '@type': 'type.googleapis.com/envoy.extensions.filters.http.router.v3.Router',
-                  },
-                },
-              ]),
-              http_protocol_options: {
-                accept_http_10: false,
-              },
-              request_timeout: '604800s',  // Necessary to allow long file uploads.
-              stream_idle_timeout: '600s',  // Only completely idle streams are dropped after this timeout.
-              route_config: {
-                virtual_hosts: [
-                  {
-                    domains: [
-                      '*',
-                    ],
-                    name: 'any',
-                    routes: routes,
-                    retry_policy: {
-                      retry_on: 'connect-failure',
-                      num_retries: 4,
-                      host_selection_retry_max_attempts: 4,
+                  name: 'tls',
+                  sds_config: {
```

Comment:
so does SDS get deployed in a separate pod?

Reply:
xDS can happen over the network or by files containing the messages that would be sent over the network.  In the case of discovering the TLS secret, we just use files, hence `path_config_source`.

When you edit a configmap or secret in k8s, the mounted volumes actually change to reflect the update.  The `watched_directory` stuff picks up on that.  That way when something like cert-manager rotates the TLS secret, we immediately start using it without having to restart.